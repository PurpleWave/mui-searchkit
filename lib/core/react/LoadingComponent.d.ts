import { SearchkitComponent, SearchkitComponentProps } from "./SearchkitComponent";
export interface LoadingProps extends SearchkitComponentProps {
    children?: any;
}
export declare class LoadingComponent extends SearchkitComponent<LoadingProps, any> {
    render(): any;
}
