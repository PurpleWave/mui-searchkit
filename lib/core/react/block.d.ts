/**
 * Entry point
 * @param {String} name Block name
 * @return {Function}
 */
export declare function block(name?: any): any;
