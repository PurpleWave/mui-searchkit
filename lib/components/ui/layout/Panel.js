var __extends = (this && this.__extends) || (function () {
    var extendStatics = Object.setPrototypeOf ||
        ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
        function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
var React = require("react");
var PropTypes = require("prop-types");
var core_1 = require("../../../core");
var Panel = /** @class */ (function (_super) {
    __extends(Panel, _super);
    function Panel(props) {
        var _this = _super.call(this, props) || this;
        _this.state = {
            collapsed: props.defaultCollapsed
        };
        return _this;
    }
    Panel.prototype.componentWillReceiveProps = function (nextProps) {
        if (nextProps.defaultCollapsed != this.props.defaultCollapsed) {
            this.setState({
                collapsed: nextProps.defaultCollapsed
            });
        }
    };
    Panel.prototype.toggleCollapsed = function () {
        this.setState({
            collapsed: !this.state.collapsed
        });
    };
    Panel.prototype.render = function () {
        var _a = this.props, title = _a.title, mod = _a.mod, className = _a.className, disabled = _a.disabled, children = _a.children, collapsable = _a.collapsable;
        var collapsed = collapsable && this.state.collapsed;
        var containerBlock = core_1.block(mod)
            .state({ disabled: disabled });
        var titleDiv;
        if (collapsable) {
            titleDiv = (React.createElement("div", { className: containerBlock.el("header").state({ collapsable: collapsable, collapsed: collapsed }), onClick: this.toggleCollapsed.bind(this) }, title));
        }
        else {
            titleDiv = React.createElement("div", { className: containerBlock.el("header") }, title);
        }
        return (React.createElement("div", { className: containerBlock.mix(className) },
            titleDiv,
            React.createElement("div", { className: containerBlock.el("content").state({ collapsed: collapsed }) }, children)));
    };
    Panel.propTypes = {
        title: PropTypes.string,
        disabled: PropTypes.bool,
        mod: PropTypes.string,
        className: PropTypes.string,
        collapsable: PropTypes.bool,
        defaultCollapsed: PropTypes.bool
    };
    Panel.defaultProps = {
        disabled: false,
        collapsable: false,
        defaultCollapsed: true,
        mod: "sk-panel"
    };
    return Panel;
}(React.PureComponent));
exports.Panel = Panel;
//# sourceMappingURL=Panel.js.map