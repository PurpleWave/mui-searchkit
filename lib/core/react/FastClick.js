var __extends = (this && this.__extends) || (function () {
    var extendStatics = Object.setPrototypeOf ||
        ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
        function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
var React = require("react");
var RenderComponent_1 = require("./RenderComponent");
var NormalClickComponent = /** @class */ (function (_super) {
    __extends(NormalClickComponent, _super);
    function NormalClickComponent() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    NormalClickComponent.prototype.render = function () {
        var children = this.props.children;
        return React.cloneElement(children, {
            onClick: this.props.handler
        });
    };
    return NormalClickComponent;
}(React.PureComponent));
exports.NormalClickComponent = NormalClickComponent;
var FastClickComponent = /** @class */ (function (_super) {
    __extends(FastClickComponent, _super);
    function FastClickComponent() {
        var _this = _super !== null && _super.apply(this, arguments) || this;
        _this.threshold = 20;
        return _this;
    }
    FastClickComponent.prototype.handleMouseDown = function (event) {
        if (this.supportsTouch)
            return;
        if (event.button === 0) {
            this.props.handler();
        }
    };
    FastClickComponent.prototype.cleanupTouch = function () {
        delete this.startPoint;
    };
    FastClickComponent.prototype.getSinglePoint = function (event) {
        var touches = event.changedTouches;
        if (touches.length === 1) {
            return {
                x: touches[0].pageX,
                y: touches[0].pageY
            };
        }
        return null;
    };
    FastClickComponent.prototype.handleTouchStart = function (event) {
        this.supportsTouch = true;
        this.startPoint = this.getSinglePoint(event);
    };
    FastClickComponent.prototype.pointsWithinThreshold = function (p1, p2) {
        return (Math.abs(p1.x - p2.x) < this.threshold &&
            Math.abs(p1.y - p2.y) < this.threshold);
    };
    FastClickComponent.prototype.handleTouchEnd = function (event) {
        if (this.startPoint) {
            var endPoint = this.getSinglePoint(event);
            if (this.pointsWithinThreshold(this.startPoint, endPoint)) {
                this.props.handler();
            }
            this.cleanupTouch();
        }
    };
    FastClickComponent.prototype.handleClick = function (event) {
        event.preventDefault();
    };
    FastClickComponent.prototype.render = function () {
        var children = this.props.children;
        return React.cloneElement(children, {
            onMouseDown: this.handleMouseDown.bind(this),
            onTouchStart: this.handleTouchStart.bind(this),
            onTouchEnd: this.handleTouchEnd.bind(this),
            onClick: this.handleClick.bind(this)
        });
    };
    return FastClickComponent;
}(React.PureComponent));
exports.FastClickComponent = FastClickComponent;
var FastClick = /** @class */ (function (_super) {
    __extends(FastClick, _super);
    function FastClick() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    FastClick.prototype.render = function () {
        return RenderComponent_1.renderComponent(FastClick.component, this.props, this.props.children);
    };
    FastClick.component = NormalClickComponent;
    return FastClick;
}(React.Component));
exports.FastClick = FastClick;
//# sourceMappingURL=FastClick.js.map