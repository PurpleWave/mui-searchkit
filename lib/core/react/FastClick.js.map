{"version":3,"file":"FastClick.js","sourceRoot":"","sources":["../../../src/core/react/FastClick.ts"],"names":[],"mappings":";;;;;;;;;;;AAAA,6BAA+B;AAC/B,qDAAiD;AAKjD;IAA0C,wCAA6B;IAAvE;;IAQA,CAAC;IANC,qCAAM,GAAN;QACE,IAAM,QAAQ,GAAQ,IAAI,CAAC,KAAK,CAAC,QAAQ,CAAC;QAC1C,OAAO,KAAK,CAAC,YAAY,CAAC,QAAQ,EAAE;YAClC,OAAO,EAAC,IAAI,CAAC,KAAK,CAAC,OAAO;SAC3B,CAAC,CAAA;IACJ,CAAC;IACH,2BAAC;AAAD,CAAC,AARD,CAA0C,KAAK,CAAC,aAAa,GAQ5D;AARY,oDAAoB;AAWjC;IAAwC,sCAA6B;IAArE;QAAA,qEAiEC;QA9DC,eAAS,GAAG,EAAE,CAAA;;IA8DhB,CAAC;IA3DC,4CAAe,GAAf,UAAgB,KAAK;QACnB,IAAI,IAAI,CAAC,aAAa;YAAE,OAAM;QAC9B,IAAG,KAAK,CAAC,MAAM,KAAK,CAAC,EAAC;YACpB,IAAI,CAAC,KAAK,CAAC,OAAO,EAAE,CAAA;SACrB;IACH,CAAC;IAED,yCAAY,GAAZ;QACE,OAAO,IAAI,CAAC,UAAU,CAAA;IACxB,CAAC;IAED,2CAAc,GAAd,UAAe,KAAK;QAClB,IAAI,OAAO,GAAG,KAAK,CAAC,cAAc,CAAA;QAClC,IAAG,OAAO,CAAC,MAAM,KAAK,CAAC,EAAC;YACtB,OAAO;gBACL,CAAC,EAAC,OAAO,CAAC,CAAC,CAAC,CAAC,KAAK;gBAClB,CAAC,EAAC,OAAO,CAAC,CAAC,CAAC,CAAC,KAAK;aACnB,CAAA;SACF;QACD,OAAO,IAAI,CAAA;IACb,CAAC;IAED,6CAAgB,GAAhB,UAAiB,KAAK;QACpB,IAAI,CAAC,aAAa,GAAG,IAAI,CAAA;QACzB,IAAI,CAAC,UAAU,GAAG,IAAI,CAAC,cAAc,CAAC,KAAK,CAAC,CAAA;IAC9C,CAAC;IAED,kDAAqB,GAArB,UAAsB,EAAE,EAAE,EAAE;QAC1B,OAAM,CACJ,IAAI,CAAC,GAAG,CAAC,EAAE,CAAC,CAAC,GAAG,EAAE,CAAC,CAAC,CAAC,GAAG,IAAI,CAAC,SAAS;YACtC,IAAI,CAAC,GAAG,CAAC,EAAE,CAAC,CAAC,GAAG,EAAE,CAAC,CAAC,CAAC,GAAG,IAAI,CAAC,SAAS,CACvC,CAAA;IACH,CAAC;IAED,2CAAc,GAAd,UAAe,KAAK;QAClB,IAAG,IAAI,CAAC,UAAU,EAAC;YACjB,IAAI,QAAQ,GAAG,IAAI,CAAC,cAAc,CAAC,KAAK,CAAC,CAAA;YACzC,IAAG,IAAI,CAAC,qBAAqB,CAAC,IAAI,CAAC,UAAU,EAAE,QAAQ,CAAC,EAAC;gBACvD,IAAI,CAAC,KAAK,CAAC,OAAO,EAAE,CAAA;aACrB;YACD,IAAI,CAAC,YAAY,EAAE,CAAA;SACpB;IAEH,CAAC;IAED,wCAAW,GAAX,UAAY,KAAK;QACf,KAAK,CAAC,cAAc,EAAE,CAAA;IACxB,CAAC;IAED,mCAAM,GAAN;QACE,IAAM,QAAQ,GAAQ,IAAI,CAAC,KAAK,CAAC,QAAQ,CAAC;QAC1C,OAAO,KAAK,CAAC,YAAY,CAAC,QAAQ,EAAE;YAClC,WAAW,EAAC,IAAI,CAAC,eAAe,CAAC,IAAI,CAAC,IAAI,CAAC;YAC3C,YAAY,EAAC,IAAI,CAAC,gBAAgB,CAAC,IAAI,CAAC,IAAI,CAAC;YAC7C,UAAU,EAAC,IAAI,CAAC,cAAc,CAAC,IAAI,CAAC,IAAI,CAAC;YACzC,OAAO,EAAC,IAAI,CAAC,WAAW,CAAC,IAAI,CAAC,IAAI,CAAC;SACpC,CAAC,CAAA;IACJ,CAAC;IAEH,yBAAC;AAAD,CAAC,AAjED,CAAwC,KAAK,CAAC,aAAa,GAiE1D;AAjEY,gDAAkB;AAmE/B;IAA+B,6BAAyB;IAAxD;;IAUA,CAAC;IAPC,0BAAM,GAAN;QACE,OAAO,iCAAe,CACpB,SAAS,CAAC,SAAS,EACnB,IAAI,CAAC,KAAK,EACV,IAAI,CAAC,KAAK,CAAC,QAAQ,CACpB,CAAA;IACH,CAAC;IARM,mBAAS,GAAG,oBAAoB,CAAA;IASzC,gBAAC;CAAA,AAVD,CAA+B,KAAK,CAAC,SAAS,GAU7C;AAVY,8BAAS"}