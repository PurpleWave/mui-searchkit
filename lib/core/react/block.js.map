{"version":3,"file":"block.js","sourceRoot":"","sources":["../../../src/core/react/block.ts"],"names":[],"mappings":";AAAA,IAAM,MAAM,GAAG,OAAO,CAAC,eAAe,CAAC,CAAA;AACtC,4BAA4B;AAE5B,IAAI,EAAE,GAAG,KAAK,EACb,KAAK,GAAG,GAAG,EACX,QAAQ,GAAG;IACV,EAAE,EAAE,EAAE;IACN,EAAE,EAAE,IAAI;IACR,GAAG,EAAE,GAAG;IACR,QAAQ,EAAE,GAAG;IACb,QAAQ,EAAE,EAAE;CACZ,CAAC;AAEH;;GAEG;AACH,gBAAgB,MAAM,EAAE,GAAG;IAC1B,OAAO,MAAM,CAAC,IAAI,CAAC,GAAG,CAAC,CAAC,MAAM,CAAC,UAAS,MAAM,EAAE,GAAG;QAClD,IAAI,KAAK,GAAG,GAAG,CAAC,GAAG,CAAC,CAAC;QAErB,wBAAwB;QACxB,IAAK,KAAK,CAAC,OAAO,CAAC,KAAK,CAAC,EAAG;YAC3B,KAAK,GAAG,KAAK,CAAC,KAAK,EAAE,CAAC;SACtB;QAED,MAAM,CAAC,GAAG,CAAC,GAAG,KAAK,CAAC;QACpB,OAAO,MAAM,CAAC;IACf,CAAC,EAAE,MAAM,CAAC,CAAC;AACZ,CAAC;AAED;;GAEG;AACH,cAAc,GAAG;IAChB,OAAO,MAAM,CAAC,EAAE,EAAE,GAAG,CAAC,CAAC;AACxB,CAAC;AAED;;;;;;GAMG;AACH,uBAAuB,GAAG,EAAE,SAAU;IACrC,IAAK,SAAS,KAAK,SAAS,EAAG;QAC9B,SAAS,GAAG,QAAQ,CAAC,GAAG,CAAC;KACzB;IAED,IAAI,iBAAiB,GAAG,QAAQ,CAAC,QAAQ,CAAC;IAE1C,OAAO,MAAM,CAAC,IAAI,CAAC,GAAG,CAAC,CAAC,MAAM,CAAC,UAAS,KAAK,EAAE,GAAG;QACjD,IAAI,KAAK,GAAG,GAAG,CAAC,GAAG,CAAC,CAAC;QAErB,IAAK,CAAC,KAAK,EAAG;YACb,OAAO,KAAK,CAAC;SACb;QAED,IAAK,KAAK,KAAK,IAAI,EAAG;YACrB,KAAK,CAAC,IAAI,CAAC,SAAS,GAAG,GAAG,CAAC,CAAC;SAC5B;aAAM;YACN,kDAAkD;YAClD,KAAK,CAAC,IAAI,CAAC,SAAS,GAAG,GAAG,GAAG,iBAAiB,GAAG,KAAK,CAAC,CAAC;SACxD;QAED,OAAO,KAAK,CAAC;IACd,CAAC,EAAE,EAAE,CAAC,CAAC;AACR,CAAC;AAED;;;;GAIM;AACN,0BAA0B,IAAI;IAC7B,OAAO,QAAQ,CAAC,QAAQ,CAAC,IAAI,CAAC,IAAI,IAAI,CAAC;AACxC,CAAC;AAED;;GAEG;AACH;IACC,IAAI,IAAI,GAAG,KAAK,CAAC,SAAS,CAAC,KAAK,CAAC,IAAI,CAAC,SAAS,CAAC,EAC/C,OAAO,GAAG,IAAI,CAAC,IAAI,CAAC,CAAC;IAEtB,OAAO,GAAG,IAAI,CAAC,MAAM,CAAC,UAAS,OAAO,EAAE,IAAI;QAC3C,IAAK,IAAI,IAAI,OAAO,IAAI,KAAK,QAAQ,EAAG;YACvC,OAAO,CAAC,IAAI,GAAG,OAAO,CAAC,IAAI,GAAG,QAAQ,CAAC,EAAE,GAAG,IAAI,CAAC;SACjD;QAED,IAAK,IAAI,IAAI,OAAO,IAAI,KAAK,QAAQ,EAAG;YACvC,OAAO,CAAC,IAAI,CAAC,IAAI,CAAC,IAAI,CAAC,CAAC;SACxB;QAED,OAAO,OAAO,CAAC;IAChB,CAAC,EAAE,OAAO,CAAC,CAAC;IAEZ,OAAO,OAAO,CAAC,OAAO,CAAC,CAAC;AACzB,CAAC;AAED;;GAEG;AACH;IACC,gBAAgB;IAChB,IAAI,IAAI,GAAG,QAAQ,CAAC,EAAE,GAAG,IAAI,CAAC,IAAI,EACjC,SAAS,GAAG,gBAAgB,CAAC,IAAI,CAAC,CAAC;IAEpC,gBAAgB;IAChB,SAAS,GAAG,IAAI,CAAC,IAAI,CAAC,MAAM,CAAC,UAAS,SAAS,EAAE,SAAS;QACzD,IAAI,QAAQ,GAAG,aAAa,CAAC,SAAS,CAAC,CAAC;QAExC,IAAK,QAAQ,CAAC,MAAM,EAAG;YACtB,QAAQ,GAAG,QAAQ,CAAC,GAAG,CAAC,UAAU,GAAG;gBACpC,OAAO,gBAAgB,CAAC,IAAI,GAAG,GAAG,CAAC,CAAC;YACrC,CAAC,CAAC,CAAC;YAEH,QAAQ,CAAC,OAAO,CAAC,EAAE,CAAC,CAAC;YACrB,SAAS,IAAI,QAAQ,CAAC,IAAI,CAAC,KAAK,CAAC,CAAC;SAClC;QAED,OAAO,SAAS,CAAC;IAClB,CAAC,EAAE,SAAS,CAAC,CAAC;IAEd,2BAA2B;IAC3B,IAAK,IAAI,CAAC,KAAK,CAAC,MAAM,EAAG;QACxB,SAAS,IAAI,KAAK,GAAG,IAAI,CAAC,KAAK,CAAC,IAAI,CAAC,KAAK,CAAC,CAAC;KAC5C;IAED,aAAa;IACb,IAAI,MAAM,GAAG,IAAI,CAAC,MAAM,CAAC;IACzB,SAAS,GAAG,MAAM,CAAC,IAAI,CAAC,MAAM,CAAC,CAAC,MAAM,CAAC,UAAS,SAAS,EAAE,KAAK;QAC/D,OAAO,SAAS,IAAI,MAAM,CAAC,KAAK,CAAC,CAAC,CAAC,CAAC,KAAK,GAAG,EAAE,GAAG,KAAK,CAAC,CAAC,CAAC,EAAE,CAAC;IAC7D,CAAC,EAAE,SAAS,CAAC,CAAC;IAEd,OAAO,SAAS,CAAC;AAClB,CAAC;AAED;IACC,IAAI,UAAU,GAAG,QAAQ,CAAC,IAAI,CAAC,IAAI,CAAC,EAAE,CAAC;IACvC,OAAO,MAAM,CAAC,SAAS,CAAC,KAAK,CAAC,KAAK,CAAC,UAAU,EAAE,SAAS,CAAC,CAAC;AAC5D,CAAC;AAED;;;GAGG;AACH,aAAa,SAAS;IACrB,IAAI,OAAO,GAAG,IAAI,CAAC,IAAI,CAAC,EACvB,OAAO,CAAC;IAET,IAAK,SAAS,EAAG;QAChB,IAAK,OAAO,SAAS,KAAK,UAAU,EAAG;YACtC,OAAO,GAAG,CAAE,SAAS,CAAC,QAAQ,EAAE,CAAE,CAAC;SACnC;aAAM,IAAK,KAAK,CAAC,OAAO,CAAC,SAAS,CAAC,EAAG;YACrC,OAAO,GAAG,SAAS,CAAC;SACrB;aAAM,IAAK,OAAO,SAAS,KAAK,QAAQ,EAAG;YAC1C,OAAO,GAAG,CAAE,SAAS,CAAC,QAAQ,EAAE,CAAE,CAAC;SACpC;aAAM,IAAK,OAAO,SAAS,KAAK,QAAQ,EAAG;YAC3C,OAAO,GAAG,CAAE,SAAS,CAAE,CAAC;SACxB;aAAO;YACP,OAAO,GAAG,aAAa,CAAC,SAAS,EAAE,EAAE,CAAC,CAAC;SACvC;QACD,OAAO,CAAC,KAAK,GAAG,OAAO,CAAC,KAAK,CAAC,MAAM,CAAC,OAAO,CAAC,CAAC;KAC9C;IAED,OAAO,OAAO,CAAC,OAAO,CAAC,CAAC;AACzB,CAAC;AAED;;;;GAIG;AACH,eAAe,GAAG;IACjB,IAAI,OAAO,GAAG,IAAI,CAAC,IAAI,CAAC,EACvB,MAAM,GAAG,IAAI,CAAC,OAAO,CAAC,MAAM,CAAC,CAAC;IAE/B,MAAM,CAAC,MAAM,EAAE,GAAG,IAAI,EAAE,CAAC,CAAC;IAC1B,OAAO,CAAC,MAAM,GAAG,MAAM,CAAC;IAExB,OAAO,OAAO,CAAC,OAAO,CAAC,CAAC;AACzB,CAAC;AAED;;;;GAIG;AACH,iBAAiB,OAAO;IACvB,OAAO,GAAG,MAAM,CAAC;QAChB,IAAI,EAAE,EAAE;QACR,IAAI,EAAE,EAAE;QACR,KAAK,EAAE,EAAE;QACT,MAAM,EAAE,EAAE;KACV,EAAE,OAAO,IAAI,EAAE,CAAC,CAAC;IAElB,oEAAoE;IACpE,8CAA8C;IAC9C,IAAI,CAAC,GAAO,EAAE,CAAA;IACd,CAAC,CAAC,EAAE,GAAG,gBAAgB,CAAC,IAAI,CAAC,OAAO,CAAC,CAAA;IACrC,CAAC,CAAC,GAAG,GAAG,gBAAgB,CAAC,IAAI,CAAC,OAAO,CAAC,CAAA;IACtC,iBAAiB;IACjB,CAAC,CAAC,QAAQ,GAAG,QAAQ,CAAC,IAAI,CAAC,OAAO,CAAC,CAAC;IACpC,CAAC,CAAC,KAAK,GAAG,KAAK,CAAC,IAAI,CAAC,OAAO,CAAC,CAAC;IAC9B,CAAC,CAAC,GAAG,GAAG,GAAG,CAAC,IAAI,CAAC,OAAO,CAAC,CAAC;IAC1B,CAAC,CAAC,KAAK,GAAG,KAAK,CAAC,IAAI,CAAC,OAAO,CAAC,CAAC;IAE9B,OAAO,CAAC,CAAC;AACV,CAAC;AAED;;;;GAIG;AACH,eAAsB,IAAK;IAC1B,OAAO,OAAO,CAAC,EAAE,IAAI,EAAE,IAAI,EAAE,CAAC,CAAC;AAChC,CAAC;AAFD,sBAEC;AACD;;GAEG;AACH,MAAM,CAAC,KAAK,EAAE;IACb,KAAK,EAAC,UAAS,GAAG;QACjB,MAAM,CAAC,QAAQ,EAAE,GAAG,IAAI,EAAE,CAAC,CAAC;QAC5B,OAAO,KAAK,CAAC;IACd,CAAC;CACD,CAAC,CAAC"}