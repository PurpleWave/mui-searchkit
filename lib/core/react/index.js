function __export(m) {
    for (var p in m) if (!exports.hasOwnProperty(p)) exports[p] = m[p];
}
Object.defineProperty(exports, "__esModule", { value: true });
__export(require("./block"));
__export(require("./FastClick"));
__export(require("./LoadingComponent"));
__export(require("./RenderComponent"));
__export(require("./SearchkitComponent"));
__export(require("./SearchkitProvider"));
//# sourceMappingURL=index.js.map