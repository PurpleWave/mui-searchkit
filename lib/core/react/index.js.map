{"version":3,"file":"index.js","sourceRoot":"","sources":["../../../src/core/react/index.ts"],"names":[],"mappings":";;;;AAAA,6BAAuB;AACvB,iCAA2B;AAC3B,wCAAkC;AAClC,uCAAiC;AACjC,0CAAoC;AACpC,yCAAmC"}