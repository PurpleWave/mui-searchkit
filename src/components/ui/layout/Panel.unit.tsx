import * as React from "react";
import {mount} from "enzyme";
import {Panel} from "./Panel"

describe("Panel", ()=> {
  beforeEach(()=> {
    this.wrapper = mount(
      <Panel title="My Panel">
        <p>panel content...</p>
      </Panel>
    )
  })
  it("should render correctly", ()=> {
    expect(this.wrapper).toMatchSnapshot()    
  })

  it("should be collapsable", ()=> {
    this.wrapper = mount(
      <Panel title="My Panel" collapsable={true}>
        <p>panel content...</p>
      </Panel>
    )
    expect(this.wrapper).toMatchSnapshot()    


    let expectIsCollapsed = (shouldBeCollapsed)=> {
      expect(this.wrapper.find(".sk-panel__content").hasClass("is-collapsed"))
        .toBe(shouldBeCollapsed)
      expect(this.wrapper.find(".sk-panel__header").hasClass("is-collapsed"))
        .toBe(shouldBeCollapsed)
    }

    //test collapsing
    expectIsCollapsed(true)
    this.wrapper.find(".sk-panel__header").simulate("click")
    expectIsCollapsed(false)
    this.wrapper.find(".sk-panel__header").simulate("click")
    expectIsCollapsed(true)
    this.wrapper.setProps({defaultCollapsed:false})
    expectIsCollapsed(false)
  })
  it("can be disabled", ()=> {
    expect(this.wrapper.find(".sk-panel").hasClass("is-disabled")).toBe(false)
    this.wrapper.setProps({disabled:true})
    expect(this.wrapper.find(".sk-panel").hasClass("is-disabled")).toBe(true)
  })

  it("mod + classname can be updated", ()=> {
    this.wrapper.setProps({mod:"sk-panel-updated", className:"my-custom-class"})
    expect(this.wrapper.find(".sk-panel-updated").hasClass("my-custom-class")).toBe(true)
  })

  it("defaultCollapsed", ()=> {
    this.wrapper = mount(
      <Panel title="My Panel" collapsable={true} defaultCollapsed={false}>
        <p>panel content...</p>
      </Panel>
    )

    expect(this.wrapper.find(".sk-panel__header").hasClass("is-collapsable"))
      .toBe(true)
    expect(this.wrapper.find(".sk-panel__header").hasClass("is-collapsed"))
      .toBe(false)

  })


})
