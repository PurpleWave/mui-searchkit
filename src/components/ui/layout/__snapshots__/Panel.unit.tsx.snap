// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Panel should be collapsable 1`] = `
<div class="sk-panel">
  <div class="sk-panel__header is-collapsable is-collapsed">My Panel</div>
  <div class="sk-panel__content is-collapsed">
    <p>panel content...</p>
  </div>
</div>
`;

exports[`Panel should render correctly 1`] = `
<div class="sk-panel">
  <div class="sk-panel__header">My Panel</div>
  <div class="sk-panel__content">
    <p>panel content...</p>
  </div>
</div>
`;
